api/handlers/app.go

```diff
@@ -657,6 +668,51 @@ func (h *App) getAppFeature(r *http.Request) (*routing.Response, error) {
 	}
 }
 
+func (h *App) restartInstance(r *http.Request) (*routing.Response, error) {
+	authInfo, _ := authorization.InfoFromContext(r.Context())
+	logger := logr.FromContextOrDiscard(r.Context()).WithName("handlers.app.restart-instance")
+	appGUID := routing.URLParam(r, "guid")
+	instanceID := routing.URLParam(r, "instance")
+	processType := routing.URLParam(r, "processType")
+
+	app, err := h.appRepo.GetApp(r.Context(), authInfo, appGUID)
+	if err != nil {
+		return nil, apierrors.LogAndReturn(logger, apierrors.ForbiddenAsNotFound(err), "Failed to fetch app from Kubernetes", "AppGUID", appGUID)
+
+	}
+	appProcesses, err := h.processRepo.ListProcesses(r.Context(), authInfo, repositories.ListProcessesMessage{
+		AppGUIDs:  []string{appGUID},
+		SpaceGUID: app.SpaceGUID,
+	})
+	if err != nil {
+		return nil, apierrors.LogAndReturn(logger, err, "failed to list processes for app")
+	}
+
+	process, hasProcessType := findProcessType(appProcesses, processType)
+	if !hasProcessType {
+		return nil, apierrors.LogAndReturn(logger,
+			apierrors.NewNotFoundError(nil, repositories.ProcessResourceType),
+			"app does not have required process type",
+		)
+	}
+	instance, err := strconv.Atoi(instanceID)
+	if err != nil {
+		return nil, apierrors.LogAndReturn(logger, err, "Instance is not a string", "InstanceID", instanceID)
```

Comment:
Maybe you meant to say "Instance is not an integer"?

Reply:
I wasn't sure how to phrase the problem -> i'll take your proposal :)